packages/quantic/force-app/main/default/lwc/quanticResultChildren/quanticResultChildren.js

```diff
@@ -0,0 +1,196 @@
+ * @category Result Template
+ * @slot before-children - Slot that allows rendering content before the list of children, only when children exist.
+ * @slot after-children - Slot that allows rendering content after the list of children, only when children exist.
+ * @example
+ * <c-quantic-result-children engine-id={engineId} folded-collection={foldedCollection} template-id="example-template-id" folded-result-list-controller={foldedResultListController} result-templates-manager={resultTemplatesManager}>
+ *   <div slot="before-children">Attached documents</div>
+ * </c-quantic-result-children>
+ */
+export default class QuanticResultChildren extends LightningElement {
+  /**
+   * The ID of the engine instance the component registers to.
+   * @api
+   * @type {string}
+   */
+  @api engineId;
+  /**
+   * The folded collection containing the result and its children.
+   * @api
+   * @type {FoldedCollection}
+   */
+  @api foldedCollection;
+  /**
+   * The ID of the template used to display the child results.
+   * @api
+   * @type {string}
+   */
+  @api templateId;
+  /**
+   * The folded result list controller responsible for executing the actions of the folded collection.
+   * @api
+   * @type {FoldedResultList}
+   */
+  @api foldedResultListController;
+  /**
+   * The template manager from which to get registered custom templates.
+   * @api
+   * @type {ResultTemplatesManager}
+   */
+  @api resultTemplatesManager;
+
+  labels = {
+    loadThread,
+    showThread,
+    hideThread,
+    noDocumentsRelated,
+    noMoreDocumentsRelated,
+  };
+
+  /** @type {boolean} */
+  areAllFoldedResultsLoaded = false;
+  /** @type {boolean} */
+  areFoldedResultsExpanded = false;
+  /** @type {boolean} */
+  foldedResultHasMoreChildren = true;
+  /** @type {Array<FoldedResult>} */
+  firstChildrenPartition;
+
+  get displayedChildren() {
+    if (!this.areAllFoldedResultsLoaded) {
+      return this?.foldedCollection?.children ?? [];
+    }
+    return (
+      (this.areFoldedResultsExpanded
+        ? this?.foldedCollection?.children
+        : this.firstChildrenPartition) ?? []
+    );
+  }
+
+  get hasChildren() {
+    return !!this.displayedChildren.length;
+  }
+
+  get isLoadingMoreResults() {
+    return this?.foldedCollection?.isLoadingMoreResults;
+  }
+
+  get areMoreResultsAvailable() {
+    return this?.foldedCollection?.moreResultsAvailable;
+  }
+
+  handleToggleFoldedResults() {
+    if (!this.areAllFoldedResultsLoaded) {
+      this.loadAllFoldedResults();
+    } else if (this.areFoldedResultsExpanded) {
+      this.showLessFoldedResults();
+    } else if (!this.areFoldedResultsExpanded) {
+      this.showMoreFoldedResults();
+    }
+  }
+
+  loadAllFoldedResults() {
+    this.firstChildrenPartition = this?.foldedCollection?.children;
+    this.foldedResultListController.loadCollection(this.foldedCollection);
+    this.areAllFoldedResultsLoaded = true;
+    this.areFoldedResultsExpanded = true;
+  }
```

Comment:
yeah, actually the `loadCollection` Headless function takes care of executing the request to load all the results AND sends an analytic event.
So the first time we load all the results we want to execute `loadCollection` and not execute `logShowMoreFoldedResults` to avoid logging analytics twice.